### PR TITLE
Efficiency module [13618]

### PR DESCRIPTION
**Pull request by jparisu**

### CODE REVIEW

**Comment by jparisu** [comment]
Tests are in their way, but the PR is already ready for review.

File: test/unittest/communication/payload_pool/MapPayloadPoolTest.cpp [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+namespace eprosima {
+namespace ddsrouter {
+namespace test {
+
+/**
+ * @brief Mock over MapPayloadPool implementing public access to private variables.
+ *
+ */
+class MockMapPayloadPool : public MapPayloadPool
```
Comment (rsanchez15) [inline test/unittest/communication/payload_pool/MapPayloadPoolTest.cpp:37]: Missing tests for `release` and `reserve` functions

File: include/ddsrouter/communication/payload_pool/PayloadPool.hpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+     * It increases \c release_count_ .
+     *
+     * @param payload object to free the data from
+     *
+     * @return true if everything ok
+     * @return false if something went wrong
+     *
+     * @throw \c IncosistencyException if more releases than reserves has been done
+     */
+    virtual bool release_(
```
Comment (rsanchez15) [inline include/ddsrouter/communication/payload_pool/PayloadPool.hpp:228]: Check that the `payload.data != nullptr ` to validate the output of the `empty()` function

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Comment by jparisu** [comment]
Missing some comment or documentation in docs

**Comment by jparisu** [comment]
Merge after PR: #140

File: docs/rst/notes/notes.rst [hunk 3]
```diff
@@ -78,3 +78,11 @@ This release includes the following **Examples**:
 This release includes the following **Documentation features**:
 
 * This same documentation.
+
```
Comment (rsanchez15) [inline docs/rst/notes/notes.rst:81]: Remove this as the forthcoming version section is already included in parent file.

File: include/ddsrouter/types/Data.hpp [hunk 4]
```diff
@@ -47,10 +47,15 @@ std::ostream& operator <<(
         std::ostream& os,
         const eprosima::fastrtps::rtps::octet& octet);
 
+// //! \c SerializedPayload_t to stream serializator
```
Comment (rsanchez15) [inline include/ddsrouter/types/Data.hpp:50]: Remove zombie code

File: src/cpp/types/Data.cpp [hunk 5]
```diff
@@ -32,9 +32,31 @@ std::ostream& operator <<(
...
+//     if (payload.length > 0)
+//     {
+//         os << payload.data[payload.length - 1];
+//     }
+
+//     os << "}";
+
+//     return os;
+// }
+
```
Comment (rsanchez15) [inline src/cpp/types/Data.cpp:56]: Remove

File: test/unittest/communication/payload_pool/MapPayloadPoolTest.cpp [hunk 6]
```diff
@@ -0,0 +1,351 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline test/unittest/communication/payload_pool/MapPayloadPoolTest.cpp:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/unittest/communication/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline test/unittest/communication/CMakeLists.txt:1]:
```suggestion
# Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/unittest/communication/payload_pool/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline test/unittest/communication/payload_pool/CMakeLists.txt:1]:
```suggestion
# Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

File: test/unittest/communication/payload_pool/PayloadPoolTest.cpp [hunk 9]
```diff
@@ -0,0 +1,381 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (rsanchez15) [inline test/unittest/communication/payload_pool/PayloadPoolTest.cpp:1]:
```suggestion
// Copyright 2022 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — APPROVED** [review]
LGTM 🚀

**Pull request merged**